### PR TITLE
[docs-beta] add id attribute to api sig elements to allow anchor linking [DOC-527]

### PR DESCRIPTION
**Pull request by PedramNavid**
## Summary & Motivation

This adds an `id` to the `<dt>` tags that define header topics, e.g.

```python
<dt id="dagster.asset">@dagster.asset</dt>
```

This allows anchor-linking to API objects.

## How I Tested These Changes

local test and click

## Changelog

NOCHANGELOG

### CODE REVIEW

**Comment by PedramNavid** [comment]
* **#24735** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24735?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @PedramNavid and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/sphinx/_ext/sphinx-mdx-builder/sphinxcontrib/mdxbuilder/writers/mdx.py [hunk 1]
```diff
@@ -353,7 +353,14 @@ def depart_desc(self, node: Element) -> None:
     def visit_desc_signature(self, node: Element) -> None:
         self.in_literal += 1
         self.new_state()
-        self.add_text("<dt>")
+        ids = node.get("ids")
+        if ids:
+            if len(ids) == 1:
+                self.add_text(f"<dt id='{ids[0]}'>")
+            else:
+                raise ValueError(f"Expected 1 id, got {len(ids)} for {node}")
```
Comment (PedramNavid) [inline docs/sphinx/_ext/sphinx-mdx-builder/sphinxcontrib/mdxbuilder/writers/mdx.py:361]: I ran this on our entire codebase and there was never more than one id, I raise an exception here to make it easier to debug (you can drop sphinx into a debugger with `sphinx -P`)

**Comment by cmpadden** [comment]
That's great! Confirmed the the id is present.
image

Trying a `yarn build` and there are still many broken anchor links. Here are some examples:

```
  - Broken anchor on source page path = /api/asset-checks:
     -> linking to /api/assets#dagster.AssetKey
     -> linking to /api/assets#dagster.AssetsDefinition
     -> linking to /api/assets#dagster.SourceAsset
     -> linking to /api/assets#dagster.AssetIn
     -> linking to /api/config#dagster.ConfigSchema
     -> linking to /api/ops#dagster.RetryPolicy
     -> linking to /api/assets#dagster.AutomationCondition
     -> linking to #dagster.AssetChecksDefinition (resolved as: /api/asset-checks#dagster.AssetChecksDefinition)
     -> linking to #dagster.AssetCheckSeverity (resolved as: /api/asset-checks#dagster.AssetCheckSeverity)
     -> linking to /api/assets#dagster.AssetDep
     -> linking to #dagster.AssetCheckSpec (resolved as: /api/asset-checks#dagster.AssetCheckSpec)
     -> linking to /api/schedules-sensors#dagster.SensorDefinition
```

But progress!

**Review by cmpadden — APPROVED** [review]

**Review by cmpadden — APPROVED** [review]
Confirmed build worked successfully locally with no anchor errors.

Some Looker errors were thrown, but likely unrelated to us:
```
WARNING: autodoc: failed to import function 'build_looker_asset_specs' from module 'dagster_looker'; the following exception was raised:
No module named 'looker_sdk' [autodoc.import_object]
WARNING: autodoc: failed to import class 'DagsterLookerLkmlTranslator' from module 'dagster_looker'; the following exception was raised:
No module named 'looker_sdk' [autodoc.import_object]
```

**Pull request merged**